toutv/transport.py

```diff
@@ -61,7 +61,11 @@ def set_proxies(self, proxies):
     def set_auth(self, auth):
         self._auth = auth
 
-    def _do_query_url(self, url, params={}, timeout=20):
+    def _do_query_url(self, url, params=None, retry_on_timeout=False, num_retries=0):
```

Comment:
Instead of adding complexity to this method, could you split them in one method to do one query and another to handle the retry logic?  You can look at `_download_segment` and `_download_segment_with_retry` in toutv/dl.py, it's a similar idea.  There's many ways to do this, but here's a idea:

 - Rename `_do_query_url` to `_do_one_query_url`
 - Add new method `_do_query_url` that takes a `num_tries` parameter and calls `_do_one_query_url` and handles the retry logic
- `num_tries` defaults to 1, which effectively means no retry. This way you can get rid of the retry_on_timeout boolean.
- Add `num_tries` parameters to all the other methods that wrap `_do_query_url`.